spec/models/activity_spec.rb

```diff
@@ -1813,4 +1813,208 @@ def create_report(organisation, fund)
       end
     end
   end
+
+  %w[requires_objectives? requires_gdi? requires_collaboration_type?].each do |method|
+    describe "##{method}" do
+      context "when activity is a fund" do
+        let(:activity) { build(:fund_activity) }
+
+        it "returns false" do
+          expect(activity.send(method.to_sym)).to eq(false)
+        end
+      end
+
+      context "when activity is a programme" do
+        let(:activity) { build(:programme_activity) }
+
+        it "returns true" do
+          expect(activity.send(method.to_sym)).to eq(true)
+        end
+      end
+
+      ["project", "third-party project"].each do |level|
+        factory = (level.underscore.parameterize(separator: "_") + "_activity").to_sym
```

Comment:
Shall we extract this into a helper method in this test file as we're using it in lots of these tests? (could call it something like`def activity_level_name(level)`)?